Pidgin.Bench/ParlotParsers/ParlotJsonParser.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Parlot.Fluent;
+using Pidgin.Examples.Json;
+
+using static Parlot.Fluent.Parsers;
+
+namespace Pidgin.Bench.ParlotParsers
+{
+    public class ParlotJsonParser
+    {
+        private static readonly Parlot.Fluent.Parser<IJson> Json;
+
+        static ParlotJsonParser()
+        {
+            var LBrace = Terms.Char('{');
+            var RBrace = Terms.Char('}');
+            var LBracket = Terms.Char('[');
+            var RBracket = Terms.Char(']');
+            var Colon = Terms.Char(':');
+            var Comma = Terms.Char(',');
+
+            var String = Terms.String(StringLiteralQuotes.Double);
+
+            var jsonString =
+                String
+                    .Then<IJson>(static s => new JsonString(s.ToString()));
+
+            var json = Deferred<IJson>();
+
+            var jsonArray =
+                Between(LBracket, Separated(Comma, json), RBracket)
+                    .Then<IJson>(static els => new JsonArray(els.ToImmutableArray()));
+
+            var jsonMember =
+                String.And(Colon).And(json)
```

Comment:
Can be optimized with `AndSkip` on the colon, since we don't care about the result of this parser.